src/wi_stuff.c

```diff
@@ -2218,8 +2527,43 @@ void WI_Start(wbstartstruct_t* wbstartstruct)
 {
   WI_initVariables(wbstartstruct);
 
-  exitpic = (wbs->lastmapinfo && wbs->lastmapinfo->exitpic[0]) ? wbs->lastmapinfo->exitpic : NULL;
-  enterpic = (wbs->nextmapinfo && wbs->nextmapinfo->enterpic[0]) ? wbs->nextmapinfo->enterpic : NULL;
+  exitpic = NULL;
+  enterpic = NULL;
+  animation = NULL;
+
+  if (wbs->lastmapinfo)
+  {
+      if (wbs->lastmapinfo->exitpic[0])
+      {
+          exitpic = wbs->lastmapinfo->exitpic;
+      }
+      if (wbs->lastmapinfo->exitanim[0])
+      {
+          if (!animation)
+          {
+              animation = Z_Calloc(1, sizeof(*animation), PU_LEVEL, NULL);
+          }
+          animation->interlevel_exiting =
+              WI_ParseInterlevel(wbs->lastmapinfo->exitanim);
+      }
+  }
+
+  if (wbs->nextmapinfo)
+  {
+      if (wbs->nextmapinfo->enterpic[0])
+      {
+          enterpic = wbs->nextmapinfo->enterpic;
+      }
+      if (wbs->lastmapinfo->enteranim[0])
```

Comment:
I still believe it needs to check for `wbs->nextmapinfo->...` in this specific paragraph?

Reply:
Wow, that's right, the available JSON intermission files are the same for input and output, that's why it worked before. Thanks.

Reply:
Fixed in https://github.com/fabiangreffrath/woof/pull/1833/commits/8208ffdbd38f16b47d67354830f1db7caf993211